firedrake/bcs.py

```diff
@@ -271,6 +271,9 @@ class DirichletBC(BCBase, DirichletBCMixin):
     @DirichletBCMixin._ad_annotate_init
     def __init__(self, V, g, sub_domain, method="topological"):
         super().__init__(V, sub_domain, method=method)
+        if len(V) > 1:
+            raise ValueError("Cannot apply boundary conditions on mixed spaces directly.\n"
```

Comment:
Why not?

Reply:
One needs to index the space (`V.sub(...)`) I can add that to the message. This valueerror is migrating from inside assembly to dirichletbc creation.